scripts/infra/cloud-instance.sh

```diff
@@ -189,6 +189,35 @@ ec2_calculate_instance_public_dns() {
     PUBLIC_DNS=$(echo "$PUBLIC_DNS" | xargs)
 }
 
+ec2__update_ssh_config() {
+    handle_help_and_instance_name_opts "$@"
+    ec2_calculate_instance_public_dns
+    local ssh_config="$HOME/.ssh/config"
+    # check if ${INSTANCE_NAME} exists in ssh config
+    if ! grep -q "Host ${INSTANCE_NAME}" "$ssh_config"; then
+        cat << EOF >> "$ssh_config"
+
+Host ${INSTANCE_NAME}
+    HostName ${PUBLIC_DNS}
+    User ec2-user
+    IdentityFile ${EC2_KEY_LOCATION}
+    StrictHostKeyChecking no
+    UserKnownHostsFile /dev/null
+EOF
+        echo "Added entry for ${INSTANCE_NAME} in ~/.ssh/config"
+    else
+        local temp_file
+        temp_file=$(mktemp)
+        awk -v ip="$PUBLIC_DNS" '
```

Comment:
This `awk` is not the easiest thing to read. It seems that the ssh_config is always updated regardless if `PUBLIC_DNS` actually changed, right? Can we avoid that?

Another approach I have in mind for clarity:

* update `~/.ssh/config` to add `Include ~/.ssh/config.d/*`, create `~/.ssh/config.d/` if it does not exist
* only do what the first portion of the condition does, which is creating an ssh config file and put it in `~/.ssh/config.d/`

Feel free to give it a try or not :)

Reply:
I guess that would work, too, but I probably won't update this significantly. I wrote it in a separate script at first, so I don't mind not merging it if it's too convoluted or not interesting to enough people.